plugins/scaffolder-backend-module-gitlab/src/actions/createProjectVariable.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import { ScmIntegrationRegistry } from '@backstage/integration';
+import { getToken } from '../util';
+import { Gitlab } from '@gitbeaker/node';
+
+/**
+ * Creates a `gitlab:create-project-variable` Scaffolder action.
+ *
+ * @param options - Templating configuration.
+ * @public
+ */
+export const createGitlabProjectVariable = (options: {
+  integrations: ScmIntegrationRegistry;
+}) => {
+  const { integrations } = options;
+  return createTemplateAction<{
+    repoUrl: string;
+    projectId: string | number;
+    key: string;
+    value: string;
+    variableType: string;
+    variableProtected: boolean;
+    masked: boolean;
+    raw: boolean;
+    environmentScope: string;
+    token?: string;
+  }>({
+    id: 'gitlab:create-project-variable',
```

Comment:
`gitlab:project-variable:create` or `pv`? depends what your thoughts are.

Reply:
I don't like this short notion `pv` we should name it `gitlab:project-variable:create`